### ISSUE TITLE
Рефакторинг

### ISSUE THREAD

**Issue by iggisv9t** [ods-ai-ml4sg/proj_news_viz#78]
Что можно сделать

- [ ] Проверить весь проект на pep8

- [x] В `preprocessing_tools.py` поменять `lemmatization` на `lemmatize`

- [x] Там же поменять самопальный кэш на lru_cache

- [ ] Проверить единоорбразие имён функций в проекте

- [ ] Унифицировать препроцессинг

**Comment by vtrokhymenko** [comment]
`preprocessing_tools.py` в ветке для статьи немного переделал

>Унифицировать препроцессинг

это как именно? так как есть сейчас (что 2 ф-и в одном файле) вполне ок/нормально

**Comment by vtrokhymenko** [comment]
вот [тут](https://github.com/ods-ai-ml4sg/proj_news_viz/blob/techarticle/nlp/news_viz/data_processing/preprocessing_tools.py) препроцессинг переделанный

**Comment by iggisv9t** [comment]
Это про то, что у нас не только твой перпроцессинг, но ещё в прототипе луиджи свой и в EDA свой, а должен быть один.

**Comment by Avenon** [comment]
Вот это эксперимент можно смело удалять: [exp002](https://github.com/ods-ai-ml4sg/proj_news_viz/tree/master/nlp/experiments/exp-002_Preprocessing/scripts)
Все равно сейчас занимаюсь тем, что переписывать буду под БД, плюс буду использовать  препроцессинг Владимира из [NLP](https://github.com/ods-ai-ml4sg/proj_news_viz/blob/master/nlp/news_viz/data_processing/preprocessing_tools.py)

**Comment by vtrokhymenko** [comment]
а что за Владимир то?

**Comment by Avenon** [comment]
Извини Виктор, имел ввиду тебя, не знаю почему в голове Владимир крутился. Поправил

**Comment by vtrokhymenko** [comment]
>  lru_cache

какое-то необычное имя для переменной как по мне
сейчас это `cache` и думаю что вполне ок

**Comment by iggisv9t** [comment]
`lru_cache` это не имя переменной, а декоратор, который делает кэширование. Предлагаю просто заменить велосипед на встроенную в язык штуку.
https://docs.python.org/3/library/functools.html

**Comment by vtrokhymenko** [comment]
типа просто перед функцией добавить `@lru_cache` и все?
дело в том, что ранее этого не юзал

**Comment by iggisv9t** [comment]
Да, импортировать из `functools` -- это стандартная либа, и потом декоратор перед функцией. Да я тоже не сильно давно узнал.

**Comment by vtrokhymenko** [comment]
> ru_cache

**Comment by vtrokhymenko** [comment]
> добавить `@lru_cache`

done
